### PR TITLE
UnliftIO.Directory.createFileLink

### PR DESCRIPTION
**Pull request by yaitskov**

### CODE REVIEW

**Comment by snoyberg** [comment]
The build failures show the reason: older versions of `directory` don't provide that function. I don't mind dropping that support with tighter lower bounds and fixing up the build matrix.

**Comment by yaitskov** [comment]
@snoyberg, thanks. I didn't notice that fact. I thought build was failing due master branch was broken.

File: unliftio/src/UnliftIO/Directory.hs [hunk 1]
```diff
@@ -149,6 +152,20 @@ createDirectoryIfMissing :: MonadIO m => Bool -> FilePath -> m ()
 createDirectoryIfMissing create_parents path0 =
   liftIO (D.createDirectoryIfMissing create_parents path0)
 
+#if MIN_VERSION_directory(1,3,1)
+-- | Lifted 'D.createFileLink'.
+--
+-- @since 0.2.16.0
```
Comment (snoyberg) [inline unliftio/src/UnliftIO/Directory.hs:158]: Can you add a package.yaml version bump and a ChangeLog entry? Also, it's worth noting in the docs that this requires directory 1.3.1 or later.

Comment (yaitskov) [reply unliftio/src/UnliftIO/Directory.hs:158]: I hope I got everything right.

Comment (snoyberg) [reply unliftio/src/UnliftIO/Directory.hs:158]: Everything looks good to me, and CI is green. Thank you!

**Pull request merged**